docs/using/installing/index.rst

```diff
@@ -492,14 +492,92 @@ The ZFS ``pool`` variable should match the pool name created in the ZFS section.
    "control-service":
       "hostname": "${CONTROL_NODE}"
       "port": 4524
+
+``${CONTROL_NODE}`` should be replaced with the address of the control node.
+The optional ``port`` variable is the port on the control node to connect to.
+This value must agree with the configuration for the control service telling it on what port to listen.
+
+The file must also include a ``dataset`` item.
+This selects and configures a dataset backend.
+All nodes must be configured to use the same dataset backend.
+
+.. _openstack-dataset-backend:
+
+OpenStack Block Device Backend Configuration
+............................................
+
+The OpenStack backend uses Cinder volumes as the storage for datasets.
+This backend can be used with Flocker dataset agent nodes run by OpenStack Nova.
+The configuration item to use OpenStack should look like:
+
+.. code-block:: yaml
+
+   dataset:
+       backend: "openstack"
+       region: "<region slug; for example, LON>"
+       auth_plugin: "<authentication plugin>"
+       ...
+
+Make sure that the ``region`` specified matches the region where the Flocker nodes run.
+OpenStack must be able to attach volumes created in that region to your Flocker agent nodes.
+
+Other items are typically required but vary depending on the OpenStack authentication plugin selected
+(Flocker relies on these plugins; it does not provide them itself).
+The ``rackspace`` plugin requires ``username``, ``api_key``, and ``auth_url``.
+To find the requirements for other plugins, see the appropriate documentation in the OpenStack project or provided with the plugin.
+
+.. _aws-dataset-backend:
+
+Amazon AWS / EBS Block Device Backend Configuration
+...................................................
+
+The AWS backend uses EBS volumes as the storage for datasets.
+This backend can be used when Flocker dataset agents are run on EC2 instances.
+The configuration item to use AWS should look like:
+
+.. code-block:: yaml
+
+   dataset:
+       backend: "aws"
+       region: "<region slug; for example, us-west-1>"
+       zone: "<availability zone slug; for example, us-west-1a>"
+       access_key_id: "<AWS API key identifier>"
+       secret_access_key: "<Matching AWS API key>"
+
+Make sure that the ``region`` and ``zone`` match each other and that both match the region and zone where the Flocker agent nodes run.
+AWS must be able to attach volumes created in that availability zone to your Flocker nodes.
+
+.. _zfs-dataset-backend:
+
+ZFS Peer-to-Peer Backend Configuration (EXPERIMENTAL)
+.....................................................
+
+The ZFS backend uses node-local storage and ZFS filesystems as the storage for datasets.
+The ZFS backend remains under development,
+it is not expected to operate reliably in many situations,
+and its use with any data that you cannot afford to lose is **strongly** discouraged at this time.
+This backend has no infrastructure requirements: it can run no matter where the Flocker dataset agents run.
```

Comment:
You need to install ZFSonLinux at least, and setup a pool, and do the SSH key crap... follow-up issue?